ca/certificate-authority.go

```diff
@@ -298,6 +312,26 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 		}
 	}
 
+	// Process requested extensions.  For now, the only extension we support is
+	// TLS Feature [RFC7633], i.e., "Must Staple", and for that we ignore the
+	// value provided by the client and just overwrite it with the specific value
+	// that only requires stapling.
+	//
+	// Other requested extensions are silently ignored.
+	//
+	// XXX(rlb): It might be good to add a generic mechanism for extensions, but
+	// on the other hand, we probably *don't* want to copy arbitrary client-
+	// provided data into extensions.
+	extensions := core.ExtensionsFromCSR(&csr)
+	requestedExtensions := []signer.Extension{}
```

Comment:
If we want to support other extensions in the future, append() will be more natural, and it's not expensive now.